packages/core/test/dialog/dialogTests.tsx

```diff
@@ -192,6 +192,16 @@ describe("<Dialog>", () => {
             // test existence here because id is generated
             assert.notExists(dialog.find(".no-default-if-no-title").hostNodes().prop("aria-labelledby"));
         });
+
+        it("supports ref objects attached to container", () => {
+            const containerRef = React.createRef<HTMLDivElement>();
+            mount(
+                <Dialog containerRef={containerRef} usePortal={false}>
+                    {createDialogContents()}
+                </Dialog>,
+            );
+            assert.isTrue(containerRef.current?.classList.contains(Classes.DIALOG_CONTAINER));
```

Comment:
you may need to use a setTimeout to wait for the lifecycle to complete and the ref object to be updated, like we do for button tests here:

https://github.com/palantir/blueprint/blob/1c7fa364e956073d8894edf01d1361fd0c07fa44/packages/core/test/buttons/buttonTests.tsx#L116-L120